### PR TITLE
[FIX] web, *: replace REPLACE_WITH with LINK_TO and UNLINK

### PR DESCRIPTION
**Pull request by pro-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by FrancoisGe** [comment]
staticList._discard => _unknownRecordCommands = [] ?? bug

File: addons/web/static/src/model/relational_model/static_list.js [hunk 1]
```diff
@@ -735,6 +767,10 @@ export class StaticList extends DataPoint {
         return record;
     }
 
+    _clearCommands() {
+        this._commands = [];
```
Comment (FrancoisGe) [inline addons/web/static/src/model/relational_model/static_list.js:771]: this._unknownRecordCommands = {};

File: addons/web/static/src/model/relational_model/record.js [hunk 2]
```diff
@@ -1030,6 +1030,12 @@ export class Record extends DataPoint {
             if ("id" in this.activeFields) {
                 this._values.id = records[0].id;
             }
+            for (const fieldName in this.activeFields) {
+                const field = this.fields[fieldName];
+                if (["one2many", "many2many"].includes(field.type) && !field.relatedPropertyField) {
+                    this._changes[fieldName]?._clearCommands();
```
Comment (FrancoisGe) [inline addons/web/static/src/model/relational_model/record.js:1036]: records in staticList still dirty ?

File: addons/web/static/src/model/relational_model/static_list.js [hunk 3]
```diff
@@ -369,9 +369,13 @@ export class StaticList extends DataPoint {
         return this.model.mutex.exec(() => this._sortBy(fieldName));
     }
 
-    async replaceWith(ids, { reload = false } = {}) {
+    async update(toLink, toUnlink, { reload } = {}) {
```
Comment (FrancoisGe) [inline addons/web/static/src/model/relational_model/static_list.js:372]: object with toLink and toUnlink

Comment (FrancoisGe) [reply addons/web/static/src/model/relational_model/static_list.js:372]: Find a better name for update (confusing with record.upadate

File: addons/web/static/src/model/relational_model/static_list.js [hunk 4]
```diff
@@ -478,9 +482,11 @@ export class StaticList extends DataPoint {
         });
     }
 
-    _applyCommands(commands) {
+    async _applyCommands(commands, { canAddOverLimit, reload } = {}) {
```
Comment (FrancoisGe) [inline addons/web/static/src/model/relational_model/static_list.js:485]: not async, check if change in enterprise still needed

File: addons/web/static/src/model/relational_model/static_list.js [hunk 5]
```diff
@@ -576,9 +582,26 @@ export class StaticList extends DataPoint {
...
                 case LINK: {
-                    const record = this._createRecordDatapoint({ ...command[2], id: command[1] });
-                    if (!this.limit || this.records.length < this.limit) {
+                    let record;
+                    if (command[1] in this._cache) {
+                        record = this._cache[command[1]];
+                    } else {
+                        record = this._createRecordDatapoint({ ...command[2], id: command[1] });
+                        if (!command[2]) {
+                            recordsToLoad.push(record);
```
Comment (pro-odoo) [inline addons/web/static/src/model/relational_model/static_list.js:591]: Move below

File: addons/web/static/tests/views/fields/many2many_field_tests.js [hunk 6]
```diff
@@ -1595,6 +1601,56 @@ QUnit.module("Fields", (hooks) => {
...
+            mockRPC: async (route, args) => {
+                if (args.method === "onchange") {
+                    if (!firstOnChange) {
+                        firstOnChange = true;
+                        await def;
+                    }
+                }
+            },
+        });
+        await click(target.querySelector(".o_data_row .o_list_record_remove"));
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/many2many_field_tests.js:1645]: assert.containsN(target, ".o_data_row", 4);

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**